zcash/src/public_key.rs

```diff
@@ -100,6 +116,32 @@ impl<N: ZcashNetwork> ZcashPublicKey<N> {
     pub fn to_viewing_key(&self) -> ViewingKey {
         self.0.clone()
     }
+
+    /// Returns a sprout public key corresponding to a sprout private key
+    pub fn from_sprout_spending_key(spending_key: &SproutSpendingKey) -> SproutViewingKey {
```

Comment:
Why does this function exist here? Couldn't you implement Into<SproutViewingKey> or From<SproutSpendingKey>, or move this to the impl of SproutViewingKey?